src/featureFlags.ts

```diff
@@ -84,6 +85,27 @@ const getSubmitArticleState = (flags: IFlags, user: User) => {
   return flags?.submit_article;
 };
 
+interface UpdateParam {
+  feature: string;
+  replacement: Flag['value'];
+  onCheckValidity(value: Flag['value']): boolean;
```

Comment:
The name here is a bit confusing. I'd suggest changing it to `isUserApplicable`. It should return `true` if the we should keep the flag as is, and `false` if not. So in the case of the onboarding, if the user visited before the date they shouldn't be applicable and we return false.

Reply:
Agreed, turning the value to false will be much better.